### PR TITLE
feat: add plugin tests

### PR DESCRIPTION
**Pull request by tushar-composio**
> [!IMPORTANT]
> Adds plugin tests to CI workflow and separates them in `tox` configuration, with import adjustment in `embedder.py`.
> 
>   - **CI Workflow**:
>     - Adds a new job for plugin tests in `.github/workflows/common.yml` for Python 3.10.
>   - **Code Changes**:
>     - Moves import of `SentenceTransformer` inside `Embedding.__init__()` in `embedder.py` to comply with pylint.
>   - **Testing**:
>     - Adds `test_plugins.py` to test various toolsets like `composio_openai`, `composio_crewai`, etc.
>     - Updates `tox.ini` to include a new `testenv:plugins` for running plugin tests separately.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for 78fc5b1d12375211be4933696de2cf5044454257. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: python/tests/test_tools/test_plugins.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                            "value of type string. This parameter is required.",
+                            "title": "Message Body",
+                            "type": "string",
+                        },
+                        "thread_id": {
+                            "description": "The ID of the thread to which the reply is to be "
+                            "drafted. Please provide a value of type string.",
+                            "title": "Thread Id",
+                            # TODO: this seems wrong, why both type and anyof
+                            "type": "string",
```
Comment (shreysingla11) [inline python/tests/test_tools/test_plugins.py:46]: The redundant type definition here could be simplified. Since `thread_id` is an optional string parameter, we could either:
1. Remove the `type: string` and keep only the `anyOf` definition, or 
2. Use `type: ["string", "null"]` instead

The current implementation with both `type` and `anyOf` is redundant and could cause confusion in schema validation.

File: python/tests/test_tools/test_plugins.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+from composio import Action, App
+from composio.client.enums.base import ActionData
+from composio.tools.local.base import action
+import composio_openai
```
Comment (shreysingla11) [inline python/tests/test_tools/test_plugins.py:4]: Consider grouping related imports together and removing unused ones:

```python
from composio import Action, App  # Action is unused
from composio.client.enums.base import ActionData  # ActionData is unused
from composio.tools.local.base import action

# Group plugin imports
import composio_openai
import composio_crewai  # Unused
import composio_langchain  # Unused
import composio_llamaindex  # Unused
```

This will make the imports cleaner and more maintainable.

**Comment by shreysingla11** [comment]
### Code Review Summary

The PR adds valuable test coverage for input serialization in the `get_tools()` method, particularly focusing on the OpenAI toolset implementation. The tests verify proper handling of:
- Required and optional parameters
- Parameter type definitions
- Docstring parsing
- Tool naming conventions

#### Suggestions for Improvement:
1. Clean up imports by removing unused ones and grouping related imports
2. Simplify the type definition for optional parameters to avoid redundancy
3. Consider adding more test cases for:
   - Edge cases in parameter types
   - Error scenarios
   - Other runtime implementations beyond OpenAI

Overall, this is a good addition to the test suite that will help ensure consistent parameter handling across different runtime implementations. The code quality is good with room for minor improvements.
